packages/router/src/router_module.ts

```diff
@@ -658,6 +673,10 @@ function provideEnabledBlockingInitialNavigation(): Provider {
   ];
 }
 
+function warnOfFailedInitialNavigation() {
+  console.warn('Initial navigation failed. Navigating to application root.');
```

Comment:
Do we have info on which route we've tried to navigate to? I think it'd be great to include that into the warning as well:
```suggestion
  console.warn('Initial navigation to `/path/a/b/c` failed. Navigating to application root.');
```

Reply:
The initial navigation may have included several navigation attempts because it includes redirects. Tracking the attempted router navigations is possible but would require quite a bit of code. The other option would be to depend on the internal implementation of `router.initialNavigation` and initial navigation attempt is `inject(location).path(true)`. 

The second option would mean saving the path before the `router.initialNavigationCall` and then passing it to `warnOfFailedInitialNavigation`
```
const initialNavigationUrl = NG_DEV_MODE ? inject(Location).path(true) : '';
```

Reply:
Discussed offline, there are no extra changes to the error message planned in this PR.